src/muya/lib/index.js

```diff
@@ -372,6 +378,60 @@ class Muya {
     return this.keyboard.hideAllFloatTools()
   }
 
+  /**
+   * Replace the word range with the given replacement.
+   *
+   * @param {*} line A line block reference of the line that contains the word to
+   *                 replace - must be a valid reference!
+   * @param {*} wordCursor The range of the word to replace (line: "abc >foo< abc"
+   *                       whereas `>`/`<` is start and end of `wordCursor`). This
+   *                       range is replaced by `replacement`.
+   * @param {string} replacement The replacement.
+   * @param {boolean} setCursor Shoud we update the editor cursor?
+   */
+  replaceWordInline (line, wordCursor, replacement, setCursor = false) {
+    const { start: lineStart } = line
+    const { end: lineEnd } = line
+    const { start: wordStart } = wordCursor
+    const { end: wordEnd } = wordCursor
+
+    // Validate cursor ranges.
+    if (wordStart.key !== wordEnd.key) {
+      throw new Error('Expect a single line word cursor: start.key != end.key.')
+    } else if (lineStart.key !== lineEnd.key) {
+      throw new Error('Expect a single line line cursor: start.key != end.key.')
+    } else if (wordStart.offset > wordEnd.offset) {
+      throw new Error(`Invalid word cursor offset: ${wordStart.offset} should be less ${wordEnd.offset}.`)
+    } else if (lineStart.key !== wordEnd.key) {
+      throw new Error(`Cursor mismatch: Expect the same line but got ${lineStart.key} and ${wordEnd.key}.`)
+    } else if (lineStart.block.text.length < wordEnd.offset) {
+      throw new Error('Invalid cursor: Replacement length is larger than line length.')
+    }
+
+    const { block } = lineStart
+    const { offset: left } = wordStart
+    const { offset: right } = wordEnd
+
+    // Replace word range with replacement.
+    block.text = block.text.substr(0, left) + replacement + block.text.substr(right)
+
+    // Update cursor
+    if (setCursor) {
```

Comment:
Please use `contentState.cursor = {...}`, because `cursor` is a `setter` attribute on contentState, and we do some extra work when set cursor.